src/nvim/ex_docmd.c

```diff
@@ -6788,27 +6801,51 @@ void free_cd_dir(void)
 
 #endif
 
-/*
- * Deal with the side effects of changing the current directory.
- * When "local" is TRUE then this was after an ":lcd" command.
- */
-void post_chdir(int local)
+/// Deal with the side effects of changing the current directory.
+///
+/// @param scope  Scope of the function call (editor, tab or window).
+void post_chdir(int scope)
 {
+  assert(scope >= kCdScopeEditor && scope <= MAX_CD_SCOPE);
+
+  enum cd_scope s = scope; // Treat it as an enum so switch catches unused cases.
```

Comment:
I have now moved the following to the header `ex_docmd.h`
~~~~
/// The scope of a command.
///
/// The higher a number, the deeper a scope.
typedef enum {
  kCdScopeEditor,  ///< Affects the entire instance of NeoVim.
  kCdScopeTab,     ///< Affects on tab page.
  kCdScopeWindow,  ///< Affects one window.
} CdScope;

/// Last `:cd` scopes defined.
#define MAX_CD_SCOPE  kCdScopeWindow
~~~~
and I get the following error:
~~~~
Scanning dependencies of target nvim
[ 41%] Building C object src/nvim/CMakeFiles/nvim.dir/__/__/config/auto/pathdef.c.o
[ 41%] Building C object src/nvim/CMakeFiles/nvim.dir/auto/msgpack_dispatch.c.o
[ 41%] Building C object src/nvim/CMakeFiles/nvim.dir/arabic.c.o
[ 41%] Building C object src/nvim/CMakeFiles/nvim.dir/buffer.c.o
In file included from /Users/hiphish/Developer/neovim/src/nvim/buffer.c:41:
In file included from /Users/hiphish/Developer/neovim/src/nvim/ex_cmds2.h:6:
In file included from /Users/hiphish/Developer/neovim/src/nvim/ex_docmd.h:23:
/Users/hiphish/Developer/neovim/build/include/ex_docmd.h.generated.h:45:17: error: unknown type name 'CdScope'
void post_chdir(CdScope scope);
                ^
1 error generated.
~~~